### PR TITLE
experssion:  table: standalone implement for `EvalContext` and `BuildContext`

### LINKED ISSUE #51298: support to implement `expression.EvalContext` in separate packages outside `sessionctx`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #51298

### What changed and how does it work?

After this PR, we do not need to require `sessionctx.Context` to implement `expression.EvalContext/BuildContext`. So the expression contexts can have their implementations outside the `session` package.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/table/contextimpl/table.go [hunk 1]
```diff
@@ -29,11 +29,12 @@ var _ context.AllocatorContext = &TableContextImpl{}
 // TableContextImpl is used to provide context for table operations.
 type TableContextImpl struct {
 	sessionctx.Context
+	ectx exprctx.BuildContext
```
Comment (XuHuaiyu) [inline pkg/table/contextimpl/table.go:32]: why not use ctx.exprctx, it would be more readable

**Review by XuHuaiyu — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Review by lance6716 — COMMENTED** [review]
/approve

for lightning/import part

**Review by tangenta — APPROVED** [review]
lgtm for ddl part

**Pull request merged**